File: Python/Euler015.py
#!/usr/bin/env python3
import math


def main():
    target = 20
    print(combinatoric(target * 2, target))


def combinatoric(n, r):
    numer = math.factorial(n)
    denom = math.factorial(r) * math.factorial(n - r)
    return numer // denom


if __name__ == '__main__':
    main()

File: Node/Euler015.js
#!/usr/bin/env node

function main() {
  const target = 20

  console.log(Math.round(combinatoric(target * 2, target)))
}

function combinatoric(n, r) {
  numer = factorial(n)
  denom = factorial(r) * factorial(n - r)
  return numer / denom
}

function factorial(n) {
  var p = 1
  for (var x = n; x > 1; x--) {
    p *= x
  }
  return p
}
exports.factorial = factorial

if (require.main == module) {
  main();
}
